src/filesystem/implementation.js

```diff
@@ -1678,6 +1678,31 @@ function mkdir(fs, context, path, mode, callback) {
   make_directory(context, path, callback);
 }
 
+function mkdtemp(fs, context, prefix, options, callback) { 
+  callback = arguments[arguments.length - 1];
+
+  // this function is used to generate a random character set to append 
+  // to tmp dir name to make sure it is unique
+  function generateRandom() {
```

Comment:
Let's not duplicate this code.  Instead, let's modify the original code so we can reuse it.  In `shared.js`, let's do this:

```js
function generateRandom(template) {
  return template.replace(/[xy]/g, function(c) {
    var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
    return v.toString(16);
  });
}

function guid() {
  return generateRandom('xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx').toUpperCase();
}

/**
 * Generate a string of n random characters.  Defaults to n=6.
 */ 
function randomChars(n) {
  n = n || 6;
  var template = 'x'.repeat(n);
  return generateRandom(template);
}

...

module.exports = {
  randomChars: randomChars,
  ...
}
```

Reply:
NOTE: I removed `toUpperCase()` from `generateRandom`'s code, and moved it only to `guid`.